File: LinearSearch.java
// 01-LinearSearch.java

public class LinearSearch {

    // Method to perform Linear Search
    public static int linearSearch(int[] array, int target) {

        // Traverse each element one by one
        for (int i = 0; i < array.length; i++) {

            // Check if current element is equal to target
            if (array[i] == target) {
                return i; // Return index if found
            }

        }

        // Return -1 if target is not found
        return -1;
    }

    public static void main(String[] args) {

        // Sample array
        int[] numbers = {12, 25, 8, 45, 30, 18};

        // Number to search
        int target = 45;

        // Call Linear Search method
        int result = linearSearch(numbers, target);

        // Check result
        if (result != -1) {
            System.out.println("Element found at index: " + result);
        } else {
            System.out.println("Element not found.");
        }

    }
}

File: LinearSearch.c
#include <stdio.h>

int main() {
    int arr[100];
    int n, i;
    int key;
    int found = 0;

    // Input array size
    printf("Enter the number of elements: ");
    scanf("%d", &n);

    // Input array elements
    printf("Enter %d elements:\n", n);
    for(i = 0; i < n; i++) {
        scanf("%d", &arr[i]);
    }

    // Input the element to search
    printf("Enter the element to search: ");
    scanf("%d", &key);

    // Perform Linear Search
    for(i = 0; i < n; i++) {
        if(arr[i] == key) {
            found = 1;
            printf("\nElement %d found at position %d.\n", key, i + 1);
            break;
        }
    }

    if(found == 0) {
        printf("\nElement %d not found in the array.\n", key);
    }

    return 0;
}
